src/mds/CDir.cc

```diff
@@ -2974,6 +2986,13 @@ int CDir::_next_dentry_on_set(set<dentry_key_t>& dns, bool missing_okay,
     // okay, we got a  dentry
     dns.erase(dnkey);
 
+    if (dn->get_projected_version() < scrub_infop->last_recursive.version) {
+      dout(15) << " skip dentry " << dnkey.name
+	       << ", no change since last scrub" << dendl;
+      continue;
```

Comment:
This needs to be optional (or perhaps based on time since the scrub stamp was last updated) — part of the point of scrubbing is that it actually checks the data is good.